File: e-book-repository/src/main/java/com/ftn/service/LanguageService.java
package com.ftn.service;

import com.ftn.model.dto.LanguageDTO;

import java.util.List;

/**
 * Created by Jasmina on 16/06/2018.
 */
public interface LanguageService {

    List<LanguageDTO> readAll();

    LanguageDTO create(LanguageDTO languageDTO);

    LanguageDTO update(Long id, LanguageDTO languageDTO);

    void delete(Long id);
}

File: e-book-repository/src/main/webapp/services/languageService.js
/**
 * Created by Jasmina on 18/08/2018.
 */
app.service('languageService', function($http){
    return {
        read: function(onSuccess, onError){
            $http.get('/api/languages').then(onSuccess, onError);
        },
        create: function(language, onSuccess, onError){
            $http.post('/api/languages', language).then(onSuccess, onError);
        },
        update: function(language, onSuccess, onError){
            $http.patch('/api/languages/' + language.id, language).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/languages/' + id).then(onSuccess, onError);
        }
    }
});